### DISCUSSION TITLE
How to get member internal forces by member

### DISCUSSION THREAD

**Discussion by nathanvanthof** [Dlubal-Software/RFEM_Python_Client#271] [Q&A] [answered]
I would like to retrieve the internal forces for one member. I've tried:

    results =  ResultTables.MembersInternalForces(CaseObjectType.E_OBJECT_TYPE_LOAD_CASE,
                                                  loading_no=1,
                                                  object_no=1)
                                                                               
But then I get the results for all the members. I've also tried:
 - leaving object_no empty
 - passing a string instead of an integer
 - passing different integers (5 a member I do have, 100000 a member I don't have)
 
It doesn't seem to matter. I always get the exact same results (namely: across all the members). Anybody know how to get the internal results for one specific member?

A workable solution would also be to determine what result belongs to what member, I can filter it out myself then, but  based on this I don't think I can do that either:

>{'node_number': 2.0, 'internal_force_vy': 16518.740234375, 'location_flags': 'M', 'internal_force_mt': 2817.722900390625, 'location': 0.0, 'internal_force_my': 158209.09375, 'internal_force_mz': 42120.48828125, 'internal_force_vz': -32104.109375, 'internal_force_n': -83169.2890625, 'specification': 'Beam | 1 - HE 320 A | L : 5000 mm'}
{'internal_force_vy': 16518.740234375, 'location_flags': 'M', 'internal_force_mt': 2817.722900390625, 'location': 0.5, 'internal_force_my': 142157.0, 'internal_force_mz': 33861.12109375, 'internal_force_vz': -32104.109375, 'internal_force_n': -83655.9921875, 'specification': 'Beam | 1 - HE 320 A | L : 5000 mm'}
{'internal_force_vy': 16518.740234375, 'location_flags': 'M', 'internal_force_mt': 2817.722900390625, 'location': 0.5, 'internal_force_my': 142157.0, 'internal_force_mz': 33861.12109375, 'internal_force_vz': -32104.109375, 'internal_force_n': -83655.9921875, 'specification': 'Beam | 1 - HE 320 A | L : 5000 mm'}
{'internal_force_vy': 16518.740234375, 'location_flags': 'E', 'internal_force_mt': 2817.722900390625, 'location': 0.714285714285714, 'internal_force_my': 135277.59375, 'internal_force_mz': 30321.390625, 'internal_force_vz': -32104.109375, 'internal_force_n': -83864.578125, 'specification': 'Beam | 1 - HE 320 A | L : 5000 mm'}
{'internal_force_vy': 16518.740234375, 'location_flags': 'M', 'internal_force_mt': 2817.722900390625, 'location': 1.0, 'internal_force_my': 126105.0, 'internal_force_mz': 25601.75, 'internal_force_vz': -32104.109375, 'internal_force_n': -84142.703125, 'specification': 'Beam | 1 - HE 320 A | L : 5000 mm'}

**Comment by OndraMichal** [comment]
Hi @nathanvanthof,
bug was reported to our development (bug 106154).

**Comment by OndraMichal** [comment]
Hi @nathanvanthof,
bug 106154 was already fixed (4.10.2023). Do you still experience this issue?

**Reply by gvandewiel86** [reply]
Hi @nathanvanthof , @OndraMichal ,

is there already more insight on how to determine what result belongs to what member?
I also asked this question in #343, so perhaps a bit double... :angel:

**Comment by heetrojivadiya** [comment, marked as answer]
> I would like to retrieve the internal forces for one member. I've tried:
> 
> ```
> results =  ResultTables.MembersInternalForces(CaseObjectType.E_OBJECT_TYPE_LOAD_CASE,
>                                               loading_no=1,
>                                               object_no=1)
> ```
> 
> But then I get the results for all the members. I've also tried:
> 
> * leaving object_no empty
> * passing a string instead of an integer
> * passing different integers (5 a member I do have, 100000 a member I don't have)
> 
> It doesn't seem to matter. I always get the exact same results (namely: across all the members). Anybody know how to get the internal results for one specific member?
> 
> A workable solution would also be to determine what result belongs to what member, I can filter it out myself then, but based on this I don't think I can do that either:
> 
> > {'node_number': 2.0, 'internal_force_vy': 16518.740234375, 'location_flags': 'M', 'internal_force_mt': 2817.722900390625, 'location': 0.0, 'internal_force_my': 158209.09375, 'internal_force_mz': 42120.48828125, 'internal_force_vz': -32104.109375, 'internal_force_n': -83169.2890625, 'specification': 'Beam | 1 - HE 320 A | L : 5000 mm'}
> > {'internal_force_vy': 16518.740234375, 'location_flags': 'M', 'internal_force_mt': 2817.722900390625, 'location': 0.5, 'internal_force_my': 142157.0, 'internal_force_mz': 33861.12109375, 'internal_force_vz': -32104.109375, 'internal_force_n': -83655.9921875, 'specification': 'Beam | 1 - HE 320 A | L : 5000 mm'}
> > {'internal_force_vy': 16518.740234375, 'location_flags': 'M', 'internal_force_mt': 2817.722900390625, 'location': 0.5, 'internal_force_my': 142157.0, 'internal_force_mz': 33861.12109375, 'internal_force_vz': -32104.109375, 'internal_force_n': -83655.9921875, 'specification': 'Beam | 1 - HE 320 A | L : 5000 mm'}
> > {'internal_force_vy': 16518.740234375, 'location_flags': 'E', 'internal_force_mt': 2817.722900390625, 'location': 0.714285714285714, 'internal_force_my': 135277.59375, 'internal_force_mz': 30321.390625, 'internal_force_vz': -32104.109375, 'internal_force_n': -83864.578125, 'specification': 'Beam | 1 - HE 320 A | L : 5000 mm'}
> > {'internal_force_vy': 16518.740234375, 'location_flags': 'M', 'internal_force_mt': 2817.722900390625, 'location': 1.0, 'internal_force_my': 126105.0, 'internal_force_mz': 25601.75, 'internal_force_vz': -32104.109375, 'internal_force_n': -84142.703125, 'specification': 'Beam | 1 - HE 320 A | L : 5000 mm'}

Hello @nathanvanthof,

as @OndraMichal said that this bug is fixed and now you can get result table for specific member by defining `object_no`. Furthermore, you can check member number in results by setiing `include_base` parameter to `True`. For example: 
```py
result = ResultTables.MembersInternalForces(CaseObjectType.E_OBJECT_TYPE_LOAD_CASE, loading_no=1, object_no=0, include_base=True)
print(result[13])

output: {'no': 14.0, 'description': 1.0, 'internal_force_vz': xyz, 'internal_force_my': xyz, 'internal_force_mz': xyz, 'internal_force_n': xyz, 'internal_force_label': 'xyz', 'internal_force_mt': xyz, 'location': xyz, 'node_number': xyz, 'internal_force_vy': xyz, 'specification': 'xyz', 'location_flags': 'xyz'}
```
In output, `'description' = 1.0` represents member number. With help of this you can filter the results according to your need. For more details, you may follow question in issue #343.
